src/gui/qgsattributedialog.cpp

```diff
@@ -40,6 +39,22 @@ QgsAttributeDialog::~QgsAttributeDialog()
 
   if ( mOwnedFeature )
     delete mOwnedFeature;
+
+  saveGeometry();
+}
+
+void QgsAttributeDialog::saveGeometry()
+{
+  // WARNING!!!! Don't use QgsGui::enableAutoGeometryRestore for this dialog -- the object name
+  // is dynamic and is set to match the layer/feature combination.
+  QgsSettings().setValue( QStringLiteral( "Windows/AttributeDialog/geometry" ), QDialog::saveGeometry() );
```

Comment:
@nyalldawson are you sure we can't use enableAutoGeometryRestore  here. Pretty sure I added support for dynamic keys if not I can have a look at how.

Reply:
If anything just use the same settings wrapper that enableAutoGeometryRestore  does so we can keep the logic in one place.

Reply:
Got an example? I tried this approach originally but couldn't work out out